### PR TITLE
Tests for RequestInstitutionParent and RequestInstitutionChildren

### PR DESCRIPTION
**Pull request by ruanres**
**Feature/Bug description:** Previously, there was only one test file (request_institution_test.py) with some tests for RequestInstitutionParent and RequestInstitutionChildren.

**Solution:** I created a separated test file for each request model and added new tests on both.

**TODO:** 
Create tests for RequestInstitution
Add a verification to Institution Parent Request Collection Handler, to prevent the creation of a parent request when the child already has a confirmed parent. Currently, this verification is done only on the client.

### CODE REVIEW

File: backend/test/model_test/request_institution_children_test.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                'key': self.institution.key.urlsafe(),
+                'address': self.institution.address
+            },
+        }
+
+        maked_request = request.make()
+        for dict_key in expected_maked_request.keys():
+            expected, actual = None, None
+            if dict_key == "institution":
+                dict_institution = expected_maked_request[dict_key]
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_children_test.py]: made*

File: backend/test/model_test/request_institution_children_test.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            'type_of_invite': 'REQUEST_INSTITUTION_CHILDREN',
+            'institution_key': self.institution.key.urlsafe(),
+            'institution': {
+                'name': self.institution.name,
+                'key': self.institution.key.urlsafe(),
+                'address': self.institution.address
+            },
+        }
+
+        maked_request = request.make()
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_children_test.py]: made*

File: backend/test/model_test/request_institution_children_test.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        for dict_key in expected_maked_request.keys():
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_children_test.py]: check if it is possible to remove this for and test with self.assertEquals

File: backend/test/model_test/request_institution_children_test.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+            send_notification.assert_called_with(
+                current_institution=self.other_institution.key, 
+                sender_key=self.other_institution.admin, 
+                receiver_key=self.admin.key, 
+                notification_type=notification_type,
+                message='mocked_message'
+            )
+
+
+    @patch.object(EmailSender, 'send_email')
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_children_test.py]: You gotta mock the method from the entity imported in the model file. So, the correct would be: @patch('models.request_institution_children.RequestLinkEmailSender.send_email')

File: backend/test/model_test/request_institution_children_test.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+            'institution_child_name': self.other_institution.name,
+            'institution_child_email': self.other_institution.institutional_email,
+            'institution_requested_key': self.other_institution.key.urlsafe()
+        }
+
+        request.send_email(host='somehost')
+        send_email.assert_called_with(email_json)
+
+
+    @patch.object(EmailSender, 'send_email')
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_children_test.py]: same thing

File: backend/test/model_test/request_institution_parent_test.py [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+
+    def test_make(self):
+        """Test method make for parent institution request."""
+        request = generate_request(
+            self.other_user, self.institution,
+            self.other_institution, self.admin
+        )
+
+        expected_maked_request = {
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_parent_test.py]: made*

File: backend/test/model_test/request_institution_parent_test.py [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+            'type_of_invite': 'REQUEST_INSTITUTION_PARENT',
+            'institution_key': self.institution.key.urlsafe(),
+            'institution': {
+                'name': self.institution.name,
+                'key': self.institution.key.urlsafe(),
+                'address': self.institution.address
+            },
+        }
+
+        maked_request = request.make()
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_parent_test.py]: made*

File: backend/test/model_test/request_institution_parent_test.py [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+            notification_type = 'ACCEPT_INSTITUTION_LINK' if action == 'ACCEPT' else 'REJECT_INSTITUTION_LINK'
+            send_notification.assert_called_with(
+                current_institution=self.other_institution.key, 
+                receiver_key=self.admin.key, 
+                notification_type=notification_type,
+                message='mocked_message'
+            )
+    
+
+    @patch.object(EmailSender, 'send_email')
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_parent_test.py]: wrong mocked method.

File: backend/test/model_test/request_institution_parent_test.py [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+            'institution_child_name': child_institution.name,
+            'institution_child_email': child_institution.institutional_email,
+            'institution_requested_key': parent_institution.key.urlsafe()
+        }
+
+        request.send_email(host='somehost')
+        send_email.assert_called_with(email_json)
+
+
+    @patch.object(EmailSender, 'send_email')
```
Comment (raonismaneoto) [inline backend/test/model_test/request_institution_parent_test.py]: same thing

File: backend/test/model_test/request_institution_parent_test.py [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+
+        self.assertEqual(
+            'The requested institution has already been invited',
+            str(ex.exception),
+            'The exception message is not equal to the expected one')
+
+
+    def test_create_request_for_institution_linked(self):
+        """Test create request for a linked institution."""
+        self.other_institution.add_child(self.institution.key)
```
Comment (Luiz-FS) [inline backend/test/model_test/request_institution_parent_test.py]: Check unidirectional connection.

File: backend/test/model_test/request_institution_children_test.py (hunk 5 above)
Comment (ruanres) [reply backend/test/model_test/request_institution_children_test.py]: In both cases, the RequestLinkEmailSender call the send_email from EmailSender. So the method must be patched on this class

**Pull request merged**